### PR TITLE
Correct implementation of 'clonefrom' feature

### PR DESCRIPTION
**Pull request by CyberDem0n**
According to https://github.com/zalando/patroni/issues/48 'clonefrom'
tag should be boolean and it should be used to mark node as a suitable
for creation of a new replica from. If there are more then one such node
in the cluster (with tag clonefrom=true), one of them will be chosen
randomly.

### CODE REVIEW

File: features/cascading_replication.feature [hunk 1]
```diff
@@ -4,10 +4,10 @@ Feature: cascading replication
 Scenario: check a base backup from the replica
```
Comment (alexeyklyukin) [inline features/cascading_replication.feature]: Let's update the scenario to "check a base backup and streaming replication from a replica."

**Comment by alexeyklyukin** [comment]
:+1:

**Comment by alexeyklyukin** [comment]
We still need the functionality of cloning from a specific member or set of members, i.e. those that is located in the same region/availability zone/datacenter as the replica to create. The former clonefrom could implement it, because you could set the node to clone from explicitly.

What we can do here is to agree on a certain tag so that all nodes having this tag are considered close enough to be the clone destination for each other (iff they have also the clonefrom tag set), and provide a new callback that calls a script during startup that will gather and set arbitrary tags on the node. For instance, on AWS the callback can detect the current region and set it as a value of the tag above. This way, we can provide a vendor-independant mechanism to choose the preferred replicas to clone (and also to replicate) from.

**Comment by CyberDem0n** [comment]
I like the idea of making decisions based on some specific "tag". Actually to implement this idea there is no need to have a new callback. Generation of tags could be done from start up script, which generates `config.yaml` (either from scratch or from template).

**Comment by alexeyklyukin** [comment]
The idea behind the programmatically settable tags is that we might not know where the instance is running at the time we generate the patroni.yaml. Imagine the Spilo appliance running on top of Taupage, it will have the same configuration no matter which region we start the Spilo on, and then can detect the region via a callback script and sets its value for a "locality" tag.

**Pull request merged**